osu.Game/Screens/Edit/Timing/TapTimingControl.cs

```diff
@@ -179,6 +187,13 @@ private void load(OverlayColourProvider colourProvider, OsuColour colours)
                 if (handling.NewValue)
                     start();
             }, true);
+
+            track.BindTo(clock.Track);
+        }
+
+        protected override void LoadComplete()
+        {
+            track.ValueChanged += _ => waveformContainer.Child = new WaveformComparisonDisplay();
```

Comment:
I also find this super weird, that this is done here and not inside `WaveformComparisonDisplay` itself. Like, it's not getting the track passed in but if this line didn't exist, it'd likely fall over and fail due to an old waveform being disposed around it?

Reply:
This is true and I kinda ignored that without looking into it more. On a fifteen-minute try it *should* be possible to make `WaveformComparisonDisplay` refresh itself, but it's not a five-minute job. May require some restructuring inside the class to work correctly as there are multiple pieces to update.

Reply:
Everything is recalculated each beat pass / mouse movement, so looks like it's enough to just assign new waveforms to rows. In manual testing nothing seems broken.